lib/improver/nbhood/square_kernel.py

```diff
@@ -598,15 +598,22 @@ def run(self, cube, radius, mask_cube=None):
         grid_cells_x, grid_cells_y = (
             convert_distance_into_number_of_grid_cells(
                 cube, radius, MAX_RADIUS_IN_GRID_CELLS))
-        cubes_to_sum = (
-            self._set_up_cubes_to_be_neighbourhooded(cube, mask_cube))
-        neighbourhood_averaged_cubes = (
-            self._pad_and_calculate_neighbourhood(
-                cubes_to_sum, grid_cells_x, grid_cells_y))
-        neighbourhood_averaged_cube = (
-            self._remove_padding_and_mask(
-                neighbourhood_averaged_cubes, cubes_to_sum, cube.name(),
-                grid_cells_x, grid_cells_y))
+        result_slices = iris.cube.CubeList()
+        for cube_slice in cube.slices([cube.coord(axis='y'),
+                                       cube.coord(axis='x')]):
+            cubes_to_sum = (
```

Comment:
How is this mitigating memory errors? Apart from _set_up_cubes_to_be_neighbourhooded all the other methods seem to be slicing over y and x. Moving the slicing to this top level makes sense but then we don't really want to slice and merge all the separate underlying functions as that is a bit of a waste of time.

Reply:
It's wasn't, that was the clipping. This was to do with making a 2D mask match the cube data. After discussions that suggest we are only ever expecting to apply 2D masks (at least for the foreseeable future), I have done as you suggest and left this slicing at the top level and removed it from all the individual functions. I have checked this doesn't have impacts elsewhere. I have updated and removed tests as appropriate.